osu.Game.Rulesets.Osu/Difficulty/Evaluators/FlashlightEvaluator.cs

```diff
@@ -97,20 +112,34 @@ public static double EvaluateDifficultyOf(DifficultyHitObject current, bool hidd
                 // Invert the scaling factor to determine the true travel distance independent of circle size.
                 double pixelTravelDistance = osuSlider.LazyTravelDistance / scalingFactor;
 
-                // Reward sliders based on velocity.
-                sliderBonus = Math.Pow(Math.Max(0.0, pixelTravelDistance / osuCurrent.TravelTime - min_velocity), 0.5);
+                // Reward sliders based on cursor velocity.
+                sliderBonus = Math.Log(pixelTravelDistance / osuCurrent.TravelTime + 1);
+
+                // More cursor movement requires more memorisation.
+                sliderBonus *= osuSlider.LazyTravelDistance;
 
-                // Longer sliders require more memorisation.
-                sliderBonus *= pixelTravelDistance;
+                // Nerf slow slider velocity.
+                double sliderVelocity = osuSlider.Distance / osuCurrent.TravelTime;
+                sliderBonus *= Math.Clamp((sliderVelocity - min_velocity) / (max_velocity - min_velocity), 0, 1);
 
-                // Nerf sliders with repeats, as less memorisation is required.
-                if (osuSlider.RepeatCount > 0)
-                    sliderBonus /= (osuSlider.RepeatCount + 1);
+                // Nerf sliders the more repeats they have, as less memorisation is required.
+                sliderBonus /= 0.75 * osuSlider.RepeatCount + 1;
             }
 
-            result += sliderBonus * slider_multiplier;
+            result += Math.Pow(sliderBonus, 1.2) * slider_multiplier;
 
             return result;
         }
+
+        private static double getComboScaleFor(int combo)
+        {
+            // Taken from ModFlashlight.
+            if (combo >= 200)
+                return 125.0;
+            if (combo >= 100)
+                return 162.5;
+
+            return 200.0;
+        }
```

Comment:
This kind of thing goes over my head. There is also a lot more to consider than just the radius that it might be worth making another PR instead (unless someone tells me otherwise)

Reply:
confused what state this was left in, if there's a reason you can't use `GetSize` directly could you explain? as apollo said, using the mod's method is going to be a requirement to support custom radius settings. if you need to do some math on the size for a final "combo scale" result that is fine but you should be calling `GetSize` first regardless

Reply:
You can't directly use `GetSize` as it's from `Flashlight` which is a `Container` used in gameplay, but I've funneled in the mod instance to ensure we support custom settings for the mod and it just generally looks a bit cleaner now.